schedule/schedule-appengine/src/main/java/com/jasify/schedule/appengine/model/payment/DefaultPaymentService.java

```diff
@@ -91,21 +98,33 @@ static PaymentService instance() {
         }
     }
 
+    private void queueCancelTask(Transaction tx, Key paymentId) {
+        // If the transaction completes add a task to cancel the shopping cart to cancel the ShoppingCart
+        ApplicationData applicationData = ApplicationData.instance();
+        String cancelTaskDelayInMillisecondsKey = PaymentService.class.getName() + ".CancelTaskDelayInMilliseconds";
+        Integer countdownMillis = applicationData.getPropertyWithDefaultValue(cancelTaskDelayInMillisecondsKey, DEFAULT_CANCEL_TASK_DELAY_IN_MILLISECONDS);
```

Comment:
Converted this specific case to a long and modified the setProperty method to throw IllegalArgumentException if an unhandled non null value is passed in.